command/v3/v3_unshare_service_command_test.go

```diff
@@ -0,0 +1,233 @@
+				BeforeEach(func() {
+					fakeActorV2.GetSharedToSpaceGUIDReturns(
+						"",
+						v2action.Warnings{"get-shared-to-space-guid-warning"},
+						errors.New("an error"))
+				})
+
+				It("returns the error and displays all warnings", func() {
+					Expect(executeErr).To(MatchError("an error"))
+					Expect(testUI.Err).To(Say("get-shared-to-space-guid-warning"))
+				})
+			})
+
+			Context("when using the currently targeted org", func() {
+				BeforeEach(func() {
+					cmd.SpaceName = "some-space"
+					fakeActorV2.GetSharedToSpaceGUIDReturns(
+						"shared-to-space-guid",
+						v2action.Warnings{"get-shared-to-space-guid-warning"},
+						nil)
+				})
+
+				Context("when the unsharing is successful", func() {
+					BeforeEach(func() {
+						fakeActor.UnshareServiceInstanceFromSpaceReturns(
+							v3action.Warnings{"unshare-service-warning"},
+							nil)
+					})
+
+					It("unshares the service instance with the provided space and displays all warnings", func() {
+						Expect(executeErr).ToNot(HaveOccurred())
+
+						Expect(testUI.Out).To(Say("Unsharing service instance some-service-instance from org some-org / space some-space as some-user\\.\\.\\."))
+						Expect(testUI.Out).To(Say("OK"))
+						Expect(testUI.Err).To(Say("get-shared-to-space-guid-warning"))
+						Expect(testUI.Err).To(Say("unshare-service-warning"))
+
+						Expect(fakeActor.UnshareServiceInstanceFromSpaceCallCount()).To(Equal(1))
+						serviceInstanceNameArg, sourceSpaceGUID, sharedToSpaceGUID := fakeActor.UnshareServiceInstanceFromSpaceArgsForCall(0)
+						Expect(serviceInstanceNameArg).To(Equal("some-service-instance"))
+						Expect(sourceSpaceGUID).To(Equal("some-space-guid"))
+						Expect(sharedToSpaceGUID).To(Equal("shared-to-space-guid"))
+					})
+				})
+
+				Context("when the unsharing is unsuccessful", func() {
+					BeforeEach(func() {
+						fakeActor.UnshareServiceInstanceFromSpaceReturns(
+							v3action.Warnings{"unshare-service-warning"},
+							errors.New("unsharing failed"))
+					})
+
+					It("does not unshare the service instance with the provided space and displays all warnings", func() {
+						Expect(executeErr).To(MatchError("unsharing failed"))
+
+						Expect(testUI.Err).To(Say("get-shared-to-space-guid-warning"))
+						Expect(testUI.Err).To(Say("unshare-service-warning"))
+					})
+				})
+			})
+
+			Context("when using a specified org", func() {
+				BeforeEach(func() {
+					cmd.SpaceName = "some-space"
+					cmd.OrgName = "some-other-org"
+					fakeActorV2.GetSharedToSpaceGUIDReturns(
+						"shared-to-space-guid",
+						v2action.Warnings{"get-shared-to-space-guid-warning"},
+						nil)
+				})
+
+				Context("when the unsharing is successful", func() {
+					BeforeEach(func() {
+						fakeActor.UnshareServiceInstanceFromSpaceReturns(
+							v3action.Warnings{"unshare-service-warning"},
+							nil)
+					})
+
+					It("unshares the service instance with the provided space and org and displays all warnings", func() {
```

Comment:
Please check GetSharedToSpaceGUID call count and args passed in on the first call

Reply:
Thanks, added a new test for this.

Reply:
@ablease I forgot to add the reason why, it's because if we don't check call count first and the method is not called, then we will get a hard panic in the code.